recbole/model/knowledge_aware_recommender/kgin.py

```diff
@@ -0,0 +1,431 @@
+            entity_emb, user_emb = self.convs[i](
+                entity_emb, user_emb, latent_embedding, edge_index, edge_type, interact_mat, self.weight,
+                self.disen_weight_att
+            )
+            """message dropout"""
+            if mess_dropout:
+                entity_emb = self.dropout(entity_emb)
+                user_emb = self.dropout(user_emb)
+            entity_emb = F.normalize(entity_emb)
+            user_emb = F.normalize(user_emb)
+            """result emb"""
+            entity_res_emb = torch.add(entity_res_emb, entity_emb)
+            user_res_emb = torch.add(user_res_emb, user_emb)
+
+        return entity_res_emb, user_res_emb, cor
+
+
+class KGIN(KnowledgeRecommender):
+    r"""KGIN is a knowledge-aware recommendation model. It combines knowledge graph and the user-item interaction
+    graph to a new graph called collaborative knowledge graph (CKG). This model explores intents behind a user-item
+    interaction by using auxiliary item knowledge.
+    """
+
+    input_type = InputType.PAIRWISE
+
+    def __init__(self, config, dataset):
+        super(KGIN, self).__init__(config, dataset)
+
+        # load parameters info
+        self.embedding_size = config['embedding_size']
+        self.n_factors = config['n_factors']
+        self.context_hops = config['context_hops']
+        self.decay = config['l2']
+        self.sim_decay = config['sim_regularity']
+        self.node_dropout = config['node_dropout']
+        self.node_dropout_rate = config['node_dropout_rate']
+        self.mess_dropout = config['mess_dropout']
+        self.mess_dropout_rate = config['mess_dropout_rate']
+        self.ind = config['ind']
+        self.reg_weight = config['reg_weight']
+        self.temperature = config['temperature']
+
+        # define layers and loss
+        self.n_nodes = self.n_users + self.n_entities
+        self.user_embedding = nn.Embedding(self.n_users, self.embedding_size)
+        self.entity_embedding = nn.Embedding(self.n_entities, self.embedding_size)
+        self.latent_embedding = nn.Embedding(self.n_factors, self.embedding_size)
+        self.mf_loss = BPRLoss()
+        self.reg_loss = EmbLoss()
+        self.restore_user_e = None
+        self.restore_entity_e = None
+
+        # load dataset info
+        self.kg_graph = dataset.kg_graph(form='coo', value_field='relation_id')
+        self.triplets = zip(self.kg_graph.row, self.kg_graph.data, self.kg_graph.col)
+        self.interaction_feat = dataset.dataset.inter_feat
+        self.interactions = zip(
+            self.interaction_feat[self.USER_ID].numpy(), self.interaction_feat[self.ITEM_ID].numpy()
+        )
+        self.graph, self.relation_dict = self.init_graph(self.interactions, self.triplets)
+        self.edge_index, self.edge_type = self._get_edges(self.graph)
+        self.adj_mat_list, self.mean_mat_list = self.init_sparse_relational_graph(self.relation_dict)
+        self.adj_mat = self.mean_mat_list[0]
```

Comment:
`self.adj_mat_list` is defined but not used and `self.mean_mat_list` is only used to assign value to `self.adj_mat`. 
Therefore, `self.init_sparse_relational_graph()` can be simplified.